src/vs/workbench/contrib/remote/browser/remoteConnectionHealth.ts

```diff
@@ -0,0 +1,185 @@
+		@IBannerService private readonly bannerService: IBannerService,
+		@IDialogService private readonly dialogService: IDialogService,
+		@IOpenerService private readonly openerService: IOpenerService,
+		@IHostService private readonly hostService: IHostService,
+		@IStorageService private readonly storageService: IStorageService,
+		@IProductService private readonly productService: IProductService,
+	) {
+		if (this._environmentService.remoteAuthority) {
+			this._checkInitialRemoteConnectionHealth();
+		}
+	}
+
+	private async _confirmConnection(): Promise<boolean> {
+		const enum ConnectionChoice {
+			Allow = 1,
+			LearnMore = 2,
+			Cancel = 0
+		}
+
+		const { result, checkboxChecked } = await this.dialogService.prompt<ConnectionChoice>({
+			type: Severity.Warning,
+			message: localize('unsupportedGlibcWarning', "You are about to connect to an OS that is unsupported by {0}", this.productService.nameLong),
+			buttons: [
+				{
+					label: localize({ key: 'allow', comment: ['&& denotes a mnemonic'] }, "&&Allow"),
+					run: () => ConnectionChoice.Allow
+				},
+				{
+					label: localize({ key: 'learnMore', comment: ['&& denotes a mnemonic'] }, "&&Learn More"),
+					run: async () => { await this.openerService.open('https://aka.ms/vscode-remote/faq/old-linux'); return ConnectionChoice.LearnMore; }
+				}
+			],
+			cancelButton: {
+				run: () => ConnectionChoice.Cancel
+			},
+			checkbox: {
+				label: localize('remember', "Do not ask me again"),
+			}
+		});
+
+		if (result === ConnectionChoice.LearnMore) {
+			return await this._confirmConnection();
+		}
+
+		const allowed = result === ConnectionChoice.Allow;
+		if (checkboxChecked) {
+			this.storageService.store(REMOTE_UNSUPPORTED_CONNECTION_CHOICE_KEY, allowed, StorageScope.PROFILE, StorageTarget.MACHINE);
+		}
+
+		return allowed;
+	}
+
+	private async _checkInitialRemoteConnectionHealth(): Promise<void> {
+		try {
+			const environment = await this._remoteAgentService.getRawEnvironment();
+
+			if (environment && environment.isUnsupportedGlibc) {
+				let allowed = this.storageService.getBoolean(REMOTE_UNSUPPORTED_CONNECTION_CHOICE_KEY, StorageScope.PROFILE);
+				if (allowed === undefined) {
+					allowed = await this._confirmConnection();
+				}
+				if (allowed) {
+					const actions = [
+						{
+							label: localize('unsupportedGlibcBannerLearnMore', "Learn More"),
+							href: 'https://aka.ms/vscode-remote/faq/old-linux'
+						}
+					];
+					this.bannerService.show({
+						id: 'unsupportedGlibcWarning.banner',
+						message: localize('unsupportedGlibcWarning.banner', "You are connected to an OS that is unsupported by {0}.", this.productService.nameLong),
+						actions,
+						icon: Codicon.warning,
+						disableCloseAction: true
+					});
+				} else {
+					const connection = this._remoteAgentService.getConnection();
+					connection?.dispose();
+					this.hostService.openWindow({ forceReuseWindow: true, remoteAuthority: null });
```

Comment:
I am not sure if the connection should be explicitly disposed here, @alexdima PTAL

Reply:
I don't think disposing explicitly here is necessary, the connection will get disposed during the workbench lifecycle https://github.com/microsoft/vscode/blob/846090c9893272d1473525a2042ed26b64b9eb42/src/vs/workbench/services/extensions/common/abstractExtensionService.ts#L184-L192

Reply:
Addressed in https://github.com/microsoft/vscode/pull/204377/commits/7bcd6396100da57eeaf623dbde59de973be0554f